WordPress/gutenberg
# Template part block: Add category panel

## Description
Add category panel to template part block. This allows us to change the assigned area of the template part.

Related: https://github.com/WordPress/gutenberg/issues/29152
## How has this been tested?
1. Open site editor
2. Insert template part
3. Select template part block
4. Open settings sidebar
5. Switch to Block tab
6. You should see Category panel
7. Dropdown should show the correct value
8. Changing the value and saving should work

## Screenshots 
![image](https://user-images.githubusercontent.com/2256104/108689230-20266d00-74f9-11eb-91e2-9169dde24129.png)


## Types of changes
New feature (non-breaking change which adds functionality) 

## Checklist:
